### PR TITLE
header and footer component created and tests created and passing

### PR DESCRIPTION
**Pull request by Louie-cpu**

### CODE REVIEW

**Comment by Louie-cpu** [comment]
https://trello.com/c/y2krGdQt

File: src/App.test.js [hunk 1]
```diff
@@ -2,5 +2,5 @@ import { render, screen } from "@testing-library/react"
 import App from "./App"
 
 test("renders learn react link", () => {
-  render(<App />)
+  // render(<App />)
```
Comment (sjproctor) [inline src/App.test.js]: I would take a minute and figure out how you can add a valid test here. App.js mostly contains code logic. But it is rendering your header and footer components. So you can create a test that asserts against DOM elements in those two components.

File: src/components/NavButton.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import React from "react"
+import { Link } from "react-router-dom"
+const NavButton = ({ url, buttonContent }) => {
+  return
+  ;<Link to={url}>
+    <button className="nav-button">{buttonContent}</button>
+  </Link>
```
Comment (sjproctor) [inline src/components/NavButton.js]: Looks like you are missing some parentheses here.

**Review by sjproctor — COMMENTED** [review]
couple things to look at.

File: src/App.css [hunk 3]
```diff
@@ -0,0 +1,87 @@
+/* General Styles */
+* {
```
Comment (sjproctor) [inline src/App.css:2]: Looks like you have some very large tabs in this file. Make sure the settings in VS code are 2 space tabs!

File: src/App.test.js [hunk 4]
```diff
@@ -1,6 +1,18 @@
...
+		</BrowserRouter>
+	)
+
+	const headerLogoLink = screen.getByRole("link", { name: "flat finders logo" })
+	expect(headerLogoLink).toBeInTheDocument()
+
+	const footerLogoLink = screen.getByRole("link", {
+		name: "roof tops with sun behind it",
+	})
+	expect(footerLogoLink).toBeInTheDocument()
```
Comment (sjproctor) [inline src/App.test.js:17]: The indentation in this file needs some love.

File: src/__tests__/NavButton.test.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+		</BrowserRouter>
+	)
+
+	const navButton = screen.getByText("Button")
+	expect(navButton).toBeInTheDocument()
+	expect(screen.getByRole("link", { name: "Button" })).toHaveAttribute(
+		"href",
+		"/test"
+	)
+})
```
Comment (sjproctor) [inline src/__tests__/NavButton.test.js:19]: The indentation in this file needs some love.

File: src/__tests__/SearchBar.test.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+test("SearchBar", () => {
+	render(
+		<BrowserRouter>
+			<SearchBar />
+		</BrowserRouter>
+	)
+
+	const searchBar = screen.getByPlaceholderText("search apartments")
+	expect(searchBar).toBeInTheDocument()
+})
```
Comment (sjproctor) [inline src/__tests__/SearchBar.test.js:15]: The indentation in this file needs some love.

File: src/components/Footer.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+						src={Logo}
+						alt="roof tops with sun behind it"
+					/>
+				</NavLink>
+			</div>
+		</div>
+	)
+}
+
+export default Footer
```
Comment (sjproctor) [inline src/components/Footer.js:26]: The indentation in this file needs some love.

File: src/components/Header.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+			</div>
+			<div className="header-navlink-right">
+				<SearchBar />
+				<NavLink className="navlink sign-in">Sign in</NavLink>
+			</div>
+		</Nav>
+	)
+}
+
+export default Header
```
Comment (sjproctor) [inline src/components/Header.js:34]: The indentation in this file needs some love.

File: src/components/NavButton.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+import { Link } from "react-router-dom"
+const NavButton = ({ url, buttonContent }) => {
+	return (
+		<Link to={url}>
+			<button className="nav-button">{buttonContent}</button>
+		</Link>
+	)
+}
+
+export default NavButton
```
Comment (sjproctor) [inline src/components/NavButton.js:11]: The indentation in this file needs some love.

File: src/components/SearchBar.js [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+			style={BarStyling}
+			key="random1"
+			value={keyword}
+			placeholder={"search apartments"}
+			onChange={(e) => setKeyword(e.target.value)}
+		/>
+	)
+}
+
+export default SearchBar
```
Comment (sjproctor) [inline src/components/SearchBar.js:25]: The indentation in this file needs some love.

**Review by sjproctor — COMMENTED** [review]
Take a pass and clean up the indentation and I will do another review.

**Review by sjproctor — APPROVED** [review]

**Pull request merged**